### DISCUSSION TITLE
Cannot Get Twitch notifications to work.

### DISCUSSION THREAD

**Discussion by kings431** [jef/streetmerchant#1068] [Help] [answered]
I've requested api though twitch dev and setup most of the tokens but still cannot get it to work.

Does anyone have a guide on how to set this up?
Would be appriciated!

Thanks

**Comment by gmanc2** [comment]
https://twitchtokengenerator.com Im not sure how to incorporate the client secret though.

**Reply by avesanidennis** [reply]
> I've requested api though twitch dev and setup most of the tokens but still cannot get it to work.
> 
> Does anyone have a guide on how to set this up?
> Would be appriciated!
> 
> Thanks

Me too. I don't know how to set TWITCH_REFRESH_TOKEN

**Comment by jef** [comment]
Unfortunately, I don't use Twitch. Although, I believe others have it working. Perhaps one of these days I will set it up to help. Sorry :cry: 

But hopefully someone else can give you the help you're looking for!

**Comment by jef** [comment, marked as answer]
I've never used this before, but I used [these instructions](https://d-fischer.github.io/twitch-chat-client/docs/examples/basic-bot.html) to get it working. Seems to be working good!

Creating an Application:

![image](https://user-images.githubusercontent.com/12074633/104380845-a2367580-54f9-11eb-8a36-bcc1c6189cd6.png)

> Must use unique name. Use https://www.uuidgenerator.net/ to get a UUID and append it to `streetmerchant-`.

Modified Address:

![image](https://user-images.githubusercontent.com/12074633/104380907-ba0df980-54f9-11eb-96a7-d62006e6fbbc.png)

Gave me:

![image](https://user-images.githubusercontent.com/12074633/104381007-e164c680-54f9-11eb-9791-09e7fa592afb.png)

Use the `code=87xgcbqxdtfudwhii4tbt2tmr9yj3n` in this command:

```
curl --location --request POST 'https://id.twitch.tv/oauth2/token?client_id=u3qgn2z1hu2otqcp6regws89ykyvq0&client_secret=joax18druor9tpdo46807bqa1ewdgd&code=87xgcbqxdtfudwhii4tbt2tmr9yj3n&grant_type=authorization_code&redirect_uri=http://localhost'
```

Feel free to use Postman or something similar.

![image](https://user-images.githubusercontent.com/12074633/104381378-6059ff00-54fa-11eb-903c-f17891634a90.png)

With all my tokens now received, I can fill out the required information in my dotenv:

```dosini
SHOW_ONLY_SERIES=test:series
STORES=amazon
HEADLESS=true
TWITCH_CLIENT_ID=u3qgn2z1hu2otqcp6regws89ykyvq0
TWITCH_CLIENT_SECRET=joax18druor9tpdo46807bqa1ewdgd
TWITCH_ACCESS_TOKEN=s38jey4reo3n7gqct0bdvz6lr9ogp3
TWITCH_REFRESH_TOKEN=p6dsl7j8txjaaxc87dvxie0xn8i6xqcdtdqtqnk5kldrf8j031
TWITCH_CHANNEL=hijef
```

---

![image](https://user-images.githubusercontent.com/12074633/104381981-5b497f80-54fb-11eb-8043-0f2287c77621.png)

**Reply by jef** [reply]
Created a [wiki page](https://github.com/jef/streetmerchant/wiki/Notification:-Twitch) from this discussion and [updated the site](https://jef.codes/streetmerchant/reference/notification/#twitch) to include a link.

**Reply by avesanidennis** [reply]
Wow, it worked. Thank you very much. I'm so stupid, I didn't remember to update the CLIENT_SECRET ;)

**Reply by jef** [reply]
Of course! Nope, it's ok, it's kind of hard to follow if you don't have the original instructions.

Good luck ☺️
